cmd/access/access_node_builder.go

```diff
@@ -132,24 +147,225 @@ type FlowAccessNodeBuilder struct {
+		staking := signature.NewAggregationVerifier(encoding.ConsensusVoteTag)
+		beacon := signature.NewThresholdVerifier(encoding.RandomBeaconTag)
+		merger := signature.NewCombiner(encodable.ConsensusVoteSigLen, encodable.RandomBeaconSigLen)
+
+		// initialize the verifier for the protocol consensus
+		verifier := verification.NewCombinedVerifier(builder.Committee, staking, beacon, merger)
+
+		followerCore, err := consensus.NewFollower(node.Logger, builder.Committee, node.Storage.Headers, final, verifier,
+			builder.FinalizationDistributor, node.RootBlock.Header, node.RootQC, builder.Finalized, builder.Pending)
+		if err != nil {
+			return nil, fmt.Errorf("could not initialize follower core: %w", err)
+		}
+		builder.FollowerCore = followerCore
+
+		return builder.FollowerCore, nil
+	})
+
+	return builder
+}
+
+func (builder *FlowAccessNodeBuilder) buildFollowerEngine() *FlowAccessNodeBuilder {
+	builder.Component("follower engine", func(_ cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		// initialize cleaner for DB
+		cleaner := storage.NewCleaner(node.Logger, node.DB, metrics.NewCleanerCollector(), flow.DefaultValueLogGCFrequency)
+		conCache := buffer.NewPendingBlocks()
+
+		followerEng, err := follower.New(
+			node.Logger,
+			node.Network,
+			node.Me,
+			node.Metrics.Engine,
+			node.Metrics.Mempool,
+			cleaner,
+			node.Storage.Headers,
+			node.Storage.Payloads,
+			builder.FollowerState,
+			conCache,
+			builder.FollowerCore,
+			builder.SyncCore,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("could not create follower engine: %w", err)
+		}
+		builder.FollowerEng = followerEng
+
+		return builder.FollowerEng, nil
+	})
+
+	return builder
+}
+
+func (builder *FlowAccessNodeBuilder) buildFinalizedHeader() *FlowAccessNodeBuilder {
+	builder.Component("finalized snapshot", func(_ cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		finalizedHeader, err := synceng.NewFinalizedHeaderCache(node.Logger, node.State, builder.FinalizationDistributor)
+		if err != nil {
+			return nil, fmt.Errorf("could not create finalized snapshot cache: %w", err)
+		}
+		builder.FinalizedHeader = finalizedHeader
+
+		return builder.FinalizedHeader, nil
+	})
+
+	return builder
+}
+
+func (builder *FlowAccessNodeBuilder) buildSyncEngine() *FlowAccessNodeBuilder {
+	builder.Component("sync engine", func(_ cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		sync, err := synceng.New(
+			node.Logger,
+			node.Metrics.Engine,
+			node.Network,
+			node.Me,
+			node.Storage.Blocks,
+			builder.FollowerEng,
+			builder.SyncCore,
+			builder.FinalizedHeader,
+			node.State,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("could not create synchronization engine: %w", err)
+		}
+		builder.SyncEng = sync
+
+		return builder.SyncEng, nil
+	})
+
+	return builder
+}
+
+func (builder *FlowAccessNodeBuilder) BuildConsensusFollower() AccessNodeBuilder {
```

Comment:
wouldn't the consensus follower of the staked AN and unstaked AN be different? The unstaked AN wouldn't have the ability to respond to sync request while the staked AN would.

Reply:
The unstaked will have the ability to respond to sync requests, eventually. The way we can disable this for now is at the networking level, ie not letting the unstaked nodes talk to each other at all. We don't need to make any changes to the engines themselves.